### ISSUE TITLE
Addon Knobs - Add function type

### ISSUE THREAD

**Issue by AndyOGo** [storybookjs/storybook#13564]
**Is your feature request related to a problem? Please describe**
I am discovering Knobs, I see many supported types (text, bool, number...) but nothing for functions.

How to create a knobs for a function ?
`showModal={() => {}}`

I tried to pass it as object but get a console error (Invalid prop knob.value supplied)
I also tried to pass as text with "() => {}" but of course it generates an error as the component propType want a function, not a string.

**Describe the solution you'd like**
Maybe we could have something like
`showModal=func("showModal", () => {})`
And the UI would convert it with an eval, but I am afraid this eval would bring security issues. Maybe something equivalent ?

**Describe alternatives you've considered**
None.

**Are you able to assist to bring the feature to reality?**
no

**Additional context**
Storybook 6

Reopening of #9265

**Comment by shilman** [comment]
FYI, we’ve released `addon-controls` in Storybook 6.0. Controls are portable, auto-generated knobs that are intended to replace `addon-knobs`, which are slated for deprecation.

- [Controls documentation](https://storybook.js.org/docs/react/essentials/controls)
- [Migration instructions](https://github.com/storybookjs/storybook/blob/next/addons/controls/README.md#how-do-i-migrate-from-addon-knobs)

Please upgrade and try out controls today!

**Comment by AndyOGo** [comment]
@shilman 
Sounds great. Does it support functions?

**Comment by shilman** [comment]
Nope. I'm just posting this on all knobs issues to let people know which way the wind is blowing. :)

What are some use cases for a function control?

**Comment by AndyOGo** [comment]
@shilman Thanks for your quick answer and your interest.

Use cases would be costum formatters or validators (e.g. if your would use form libraries like `react-final-form`).
Render props or any functional props use to derive their values.
